components/script/dom/xrsession.rs

```diff
@@ -423,6 +428,28 @@ impl XRSession {
             "WEBXR PROFILING [raf execute]:\t{}ms",
             (time::precise_time_ns() - raf_start) as f64 / 1_000_000.
         );
+
+        self.gc_hint();
+    }
+
+    #[allow(unsafe_code)]
+    fn gc_hint(&self) {
+        let cx = *self.global().get_cx();
+        let rt = unsafe { JS_GetRuntime(cx) };
+
+        if unsafe { IsIdleGCTaskNeeded(rt) } {
+            #[cfg(feature = "xr-profile")]
+            let gc_start = time::precise_time_ns();
+
+            debug!("GC at the end of an XR rAF");
+            unsafe { JS_GC(cx, GCReason::API) };
```

Comment:
Should we use RunIdleGCTask instead? https://searchfox.org/mozilla-central/source/js/src/jsapi.cpp#1311

Reply:
Ah, that function does almost exactly what we want. It only triggers a minor gc though, but I think that's fine.

Reply:
Done.